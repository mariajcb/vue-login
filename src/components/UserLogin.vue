<template lang="html">
  <form>
    <label for="username">User Name</label>
    <input
      type="text"
      id="username"
      :value="userData.username"/>
    <label for="password">Password</label>
    <input
      type="text"
      id="password"
      :value="userData.password"/>
    <button type="button" name="button">Submit</button>
  </form>
</template>

<script>
export default {
  name: "UserLogin",
  props: {
    userData: {
      type: Object,
      required: true,
      default: () => ({})
    }
  }
};
</script>

<style lang="css" scoped></style>
