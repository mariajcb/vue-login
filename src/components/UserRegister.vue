<template lang="html">
  <div>
    <input type="text" name="username" value="" />
    <input type="text" name="user email" value="" />
    <input type="text" name="password" value="" />
    <button type="button" name="button">Submit</button>
  </div>
</template>

<script>
export default {
  name: "UserRegister"
};
</script>

<style lang="css" scoped></style>
