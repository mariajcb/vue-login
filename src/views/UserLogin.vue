<template>
    <v-container>
      <HeaderSoundsTrueLogo />
      <v-card-text class="headline">Sign in with your Sounds True Account</v-card-text>
      <user-login-form></user-login-form>
      <!-- <router-link :to="forgot-password"> -->
          Forgot Your Password?
      <!-- </router-link> -->
      <FooterCustomerService />
    </v-container>
</template>

<script>
  import HeaderSoundsTrueLogo from '@/components/HeaderSoundsTrueLogo.vue'
  import FooterCustomerService from '@/components/FooterCustomerService.vue'
  import UserLoginForm from '@/components/UserLoginForm.vue';

  export default {
    name: "UserLogin",

    components: {
      HeaderSoundsTrueLogo,
      FooterCustomerService,
      UserLoginForm,
    },

  }
</script>

<style lang="css" scoped></style>
