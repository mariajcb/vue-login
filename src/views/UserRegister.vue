<template>
    <v-container>
      <HeaderSoundsTrueLogo />
      <v-card-text class="headline">Create a free Account</v-card-text>
      <user-register-form></user-register-form>
    </v-container>
</template>

<script>
  import HeaderSoundsTrueLogo from '@/components/HeaderSoundsTrueLogo.vue'
  import UserRegisterForm from '@/components/UserRegisterForm.vue'

  export default {
    name: "UserRegister",

    components: {
      HeaderSoundsTrueLogo,
      UserRegisterForm
    },

  }
</script>

<style lang="css" scoped></style>
