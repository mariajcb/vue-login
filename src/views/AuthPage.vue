<template lang="html">
  <div>
    <UserLogin :userData="userData"/>
    <UserRegister />
  </div>
</template>

<script>
import UserLogin from "@/components/UserLogin";
import UserRegister from "@/components/UserRegister";

export default {
  name: "AuthPage",
  data() {
    return {
      userData: {
        username: '',
        password: ''
      }
    }
  },
  components: {
    UserLogin,
    UserRegister
  }
};
</script>

<style lang="css" scoped></style>
