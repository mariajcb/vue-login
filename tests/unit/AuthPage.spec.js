import { shallowMount } from "@vue/test-utils"
import AuthPage from "@/views/AuthPage"
import UserLogin from "@/components/UserLogin"
import UserRegister from "@/components/UserRegister"

describe('AuthPage', () => {
  const build = () => {
    const wrapper = shallowMount(AuthPage)

    return {
      wrapper,
      userLogin: () => wrapper.find(UserLogin),
      userRegister: () => wrapper.find(UserRegister)
    }
  }

  it('renders the component', () => {
    const { wrapper } = build()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main child components', () => {
    const { userLogin, userRegister } = build()

    expect(userLogin().exists()).toBe(true)
    expect(userRegister().exists()).toBe(true)
  })

  it('passes a binded userdata prop to UserLogin component', () => {
    const { wrapper, userLogin } = build()
    wrapper.setData({
      userData: {
        username: 'Test',
        password: '123'
      }
    })

    expect(userLogin().vm.userData).toBe(wrapper.vm.userData)
  })
})
