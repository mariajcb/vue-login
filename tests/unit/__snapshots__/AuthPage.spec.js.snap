// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage renders the component 1`] = `
<div>
  <userlogin-stub userdata="[object Object]"></userlogin-stub>
  <userregister-stub></userregister-stub>
</div>
`;
